src/utils/tests/builders/midnight/block.rs

```diff
@@ -0,0 +1,275 @@
+use crate::models::{
+	MidnightBaseTransaction, MidnightBlock, MidnightBlockDigest, MidnightBlockHeader,
+	MidnightRpcBlock, MidnightRpcTransactionEnum,
+};
+
+/// A builder for creating test Midnight blocks with default values.
+#[derive(Debug)]
+pub struct BlockBuilder {
+	header: MidnightBlockHeader,
+	body: Vec<MidnightRpcTransactionEnum>,
+	transactions_index: Vec<(String, String)>,
+}
+
+impl Default for BlockBuilder {
+	/// Default block builder with a testnet network
+	fn default() -> Self {
+		Self {
+			header: MidnightBlockHeader {
+				parent_hash: "0x0000000000000000000000000000000000000000000000000000000000000000"
+					.to_string(),
+				number: "0".to_string(),
+				state_root: "0x0000000000000000000000000000000000000000000000000000000000000000"
+					.to_string(),
+				extrinsics_root:
+					"0x0000000000000000000000000000000000000000000000000000000000000000".to_string(),
+				digest: MidnightBlockDigest { logs: vec![] },
+			},
+			body: vec![],
+			transactions_index: vec![],
+		}
+	}
+}
+
+impl BlockBuilder {
+	/// Creates a new BlockBuilder instance.
+	pub fn new() -> Self {
+		Self::default()
+	}
+
+	/// Sets the header of the block.
+	pub fn header(mut self, header: MidnightBlockHeader) -> Self {
+		self.header = header;
+		self
+	}
+
+	/// Sets the number of the block.
+	pub fn number(mut self, number: u64) -> Self {
+		self.header.number = format!("0x{:x}", number);
+		self
+	}
+
+	/// Sets the parent hash of the block.
+	pub fn parent_hash(mut self, parent_hash: String) -> Self {
+		self.header.parent_hash = parent_hash;
+		self
+	}
+
+	/// Sets the body of the block.
+	pub fn body(mut self, body: Vec<MidnightRpcTransactionEnum>) -> Self {
+		self.body = body;
+		self
+	}
+
+	/// Adds a transaction to the block.
+	pub fn add_transaction(mut self, transaction: MidnightRpcTransactionEnum) -> Self {
+		self.body.push(transaction);
+		self
+	}
+
+	/// Adds a Midnight transaction to the block.
+	pub fn add_rpc_transaction(mut self, transaction: MidnightBaseTransaction) -> Self {
+		let tx_hash = transaction.clone().tx_hash;
+		let tx_operation = MidnightRpcTransactionEnum::MidnightTransaction {
+			tx_raw: "".to_string(),
+			tx: transaction,
+		};
+
+		self.body.push(tx_operation.clone());
+
+		// TODO: Add the transaction index to the block with the serialized transaction
+		// should be of type `MidnightNodeTransaction`
```

Comment:
For now we are populating transactions with _just_ `block.body`, once we start using raw transactions from `transactions_index` we will have to somehow serialize the transactions here as well for tests not to break.